Use supported version of OpenJDK

## Describe request or inquiry 

* Upgrade the Java microservices from OpenJDK `17` to a supported version.

## Background

* According to [this page](https://jdk.java.net/17/), OpenJDK 17 is not recommended for use in production.
* [This page](https://jdk.java.net/) suggests which OpenJDK is ready for use.
* The [Bank of Anthos development guide](https://github.com/GoogleCloudPlatform/bank-of-anthos/blob/release/v0.5.5/docs/development.md#install-tools) currently suggests using OpenJDK 17.
* We are still able to download OpenJDK 17 from [this archive](https://jdk.java.net/archive/) — so I've selected a low priority for this issue.
* In [#712](https://github.com/GoogleCloudPlatform/bank-of-anthos/pull/712), we upgraded all Java services from `1.8` to `1.17` (`JDK 17`). We may have to do something similar.
* This issue was inspired by [#874](https://github.com/GoogleCloudPlatform/bank-of-anthos/issues/874).

### What purpose/environment will this feature serve? 

* This change impacts all deployments of Bank of Anthos.
* This change will reduce the vulnerability of Java services.

Looks like I ran into an issue related to this... 
Using a CloudShell terminal, I executed  "make monolith build", the builds fail.

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  46.256 s
[INFO] Finished at: 2022-09-02T15:01:32Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project ledgermonolith: Fatal error compiling: error: invalid target release: 17 -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
make: *** [Makefile:53: monolith-build] Error 1

When I update java version to version 11 in all the pom.xml files, the build completes successfully.  Maybe there's a reasonably easy way to add OpenJDK v17 to CloudShell instances; but I'm not sure the best way to proceed.

--- a/src/balancereader/pom.xml
+++ b/src/balancereader/pom.xml
@@ -161,7 +161,7 @@
                 <version>3.3.0</version>
                 <configuration>
                     <from>
-                        <image>openjdk:17-alpine</image>
+                        <image>eclipse-temurin:17-alpine</image>
                     </from>
                 </configuration>
             </plugin>

--- a/src/balancereader/pom.xml
+++ b/src/balancereader/pom.xml
@@ -161,7 +161,7 @@
                 <version>3.3.0</version>
                 <configuration>
                     <from>
-                        <image>eclipse-temurin:17-alpine</image>
+                        <image>eclipse-temurin:17-jre-alpine</image>
                     </from>
                 </configuration>
             </plugin>

--- a/src/ledgermonolith/pom.xml
+++ b/src/ledgermonolith/pom.xml
@@ -150,7 +150,7 @@
                 <version>3.3.0</version>
                 <configuration>
                     <from>
-                        <image>eclipse-temurin:17-alpine</image>
+                        <image>eclipse-temurin:17-jre-alpine</image>
                     </from>
                 </configuration>
             </plugin>

--- a/src/ledgermonolith/pom.xml
+++ b/src/ledgermonolith/pom.xml
@@ -150,7 +150,7 @@
                 <version>3.3.0</version>
                 <configuration>
                     <from>
-                        <image>openjdk:17-alpine</image>
+                        <image>eclipse-temurin:17-alpine</image>
                     </from>
                 </configuration>
             </plugin>

--- a/src/ledgerwriter/pom.xml
+++ b/src/ledgerwriter/pom.xml
@@ -151,7 +151,7 @@
                 <version>3.3.0</version>
                 <configuration>
                     <from>
-                        <image>openjdk:17-alpine</image>
+                        <image>eclipse-temurin:17-alpine</image>
                     </from>
                 </configuration>
             </plugin>

--- a/src/ledgerwriter/pom.xml
+++ b/src/ledgerwriter/pom.xml
@@ -151,7 +151,7 @@
                 <version>3.3.0</version>
                 <configuration>
                     <from>
-                        <image>eclipse-temurin:17-alpine</image>
+                        <image>eclipse-temurin:17-jre-alpine</image>
                     </from>
                 </configuration>
             </plugin>

--- a/src/transactionhistory/pom.xml
+++ b/src/transactionhistory/pom.xml
@@ -161,7 +161,7 @@
                 <version>3.3.0</version>
                 <configuration>
                     <from>
-                        <image>openjdk:17-alpine</image>
+                        <image>eclipse-temurin:17-alpine</image>
                     </from>
                 </configuration>
             </plugin>

--- a/src/transactionhistory/pom.xml
+++ b/src/transactionhistory/pom.xml
@@ -161,7 +161,7 @@
                 <version>3.3.0</version>
                 <configuration>
                     <from>
-                        <image>eclipse-temurin:17-alpine</image>
+                        <image>eclipse-temurin:17-jre-alpine</image>
                     </from>
                 </configuration>
             </plugin>

